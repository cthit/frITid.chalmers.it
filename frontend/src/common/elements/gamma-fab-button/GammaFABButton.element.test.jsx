import React from "react";
import { shallow } from "enzyme";
import GammaFABButton from "./";
import { Add } from "@material-ui/icons";

describe("<GammaFABButton/>", () => {
  test("Shallow render of primary submit <GammaFABButton/>", () => {
    const wrapper = shallow(
      <GammaFABButton primary submit onClick={() => {}} component={Add} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of secondary submit disabled <GammaFABButton/>", () => {
    const wrapper = shallow(
      <GammaFABButton secondary disabled submit component={Add} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
