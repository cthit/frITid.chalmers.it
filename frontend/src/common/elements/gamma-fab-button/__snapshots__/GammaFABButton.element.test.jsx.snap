// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaFABButton/> Shallow render of primary submit <GammaFABButton/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaFABButton
    component={[Function]}
    onClick={[Function]}
    primary={true}
    submit={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <pure(Add) />,
      "color": "primary",
      "disabled": undefined,
      "onClick": [Function],
      "type": "submit",
      "variant": "fab",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <pure(Add) />,
        "color": "primary",
        "disabled": undefined,
        "onClick": [Function],
        "type": "submit",
        "variant": "fab",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaFABButton/> Shallow render of secondary submit disabled <GammaFABButton/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaFABButton
    component={[Function]}
    disabled={true}
    secondary={true}
    submit={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <pure(Add) />,
      "color": "secondary",
      "disabled": true,
      "onClick": undefined,
      "type": "submit",
      "variant": "fab",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <pure(Add) />,
        "color": "secondary",
        "disabled": true,
        "onClick": undefined,
        "type": "submit",
        "variant": "fab",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
